File: Examples/GroupDocs.Metadata.Examples.VBasic/Utilities/PhotoCleaner.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports GroupDocs.Metadata.Tools 
Imports System.IO
Imports GroupDocs.Metadata.Formats.Image

Namespace Utilities
    'ExStart:PhotoCleaner
    Public Class PhotoCleaner
        ' absolute path to the GroupDocs.Metadata license file.
        Private Const LicensePath As String = "GroupDocs.Metadata.lic"

        ' absolute path to the photos directory.
        Public Property CleanerPath() As String
            Get
                Return m_CleanerPath
            End Get
            Set(value As String)
                m_CleanerPath = value
            End Set
        End Property
        Private m_CleanerPath As String

        Shared Sub New()
            ' set product license 
            '             * uncomment following function if you have product license
            '             * 

            'SetInternalLicense()
        End Sub

        Public Sub New(cleanerPath As String)
            ' check if directory exists
            If Not Directory.Exists(Common.MapSourceFilePath(cleanerPath)) Then
                Throw New DirectoryNotFoundException(Convert.ToString("Directory not found: ") & cleanerPath)
            End If
            ' set property
            Me.CleanerPath = cleanerPath
        End Sub
        ''' <summary>
        ''' Applies the product license
        ''' </summary>
        Private Shared Sub SetInternalLicense()
            Dim license As New License()
            license.SetLicense(LicensePath)
        End Sub
        ''' <summary>
        ''' Removes GPS data and updates the image files in a directory
        ''' </summary>
        Public Sub RemoveExifLocation()
            ' Map directory in source folder
            Dim sourceDirectoryPath As String = Common.MapSourceFilePath(Me.CleanerPath)

            ' get array of file in specific directory
            Dim files As String() = Directory.GetFiles(sourceDirectoryPath)

            For Each path As String In files
                ' get EXIF data if exists
                Dim exifMetadata As ExifMetadata = DirectCast(MetadataUtility.ExtractSpecificMetadata(path, MetadataType.EXIF), ExifMetadata)

                If exifMetadata IsNot Nothing Then
                    Dim exifInfo As ExifInfo = exifMetadata.Data

                    If exifInfo.GPSData IsNot Nothing Then
                        ' set altitude, latitude and longitude to null values
                        exifInfo.GPSData.Altitude = Nothing
                        exifInfo.GPSData.Latitude = Nothing
                        exifInfo.GPSData.LatitudeRef = Nothing
                        exifInfo.GPSData.Longitude = Nothing
                        exifInfo.GPSData.LongitudeRef = Nothing
                    End If

                    ' and update file
                    MetadataUtility.UpdateMetadata(path, exifMetadata)
                End If
            Next
            Console.WriteLine("Press any key to exit.")
        End Sub
    End Class
    'ExEnd:PhotoCleaner
End Namespace

File: Examples/GroupDocs.Metadata.Examples.CSharp/Utilities/PhotoCleaner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using GroupDocs.Metadata.Tools;
using GroupDocs.Metadata.Formats.Image;
using System.IO;

namespace GroupDocs.Metadata.Examples.Utilities.CSharp
{
    //ExStart:PhotoCleaner
    public class PhotoCleaner
    {
        // absolute path to the GroupDocs.Metadata license file.
        private const string LicensePath = @"GroupDocs.Metadata.lic";

        // absolute path to the photos directory.
        public string CleanerPath { get; set; }

        static PhotoCleaner()
        {
            /* set product license 
             * uncomment following function if you have product license
             * */
            //SetInternalLicense();
        }

        public PhotoCleaner(string cleanerPath)
        {
            // check if directory exists
            if (!Directory.Exists(Common.MapSourceFilePath( cleanerPath)))
            {
                throw new DirectoryNotFoundException("Directory not found: " + cleanerPath);
            }
            // set property
            this.CleanerPath = cleanerPath;
        }
        /// <summary>
        /// Applies the product license
        /// </summary>
        private static void SetInternalLicense()
        { 
            License license = new License();
            license.SetLicense(LicensePath);
        }
        /// <summary>
        /// Removes GPS data and updates the image files in a directory
        /// </summary>
        public void RemoveExifLocation()
        {
            // Map directory in source folder
            string sourceDirectoryPath = Common.MapSourceFilePath(this.CleanerPath);

            // get array of file in specific directory
            string[] files = Directory.GetFiles(sourceDirectoryPath);

            foreach (string path in files)
            {
                // get EXIF data if exists
                ExifMetadata exifMetadata = (ExifMetadata)MetadataUtility.ExtractSpecificMetadata(path, MetadataType.EXIF);

                if (exifMetadata != null)
                {
                    ExifInfo exifInfo = exifMetadata.Data;

                    if (exifInfo.GPSData != null)
                    {
                        // set altitude, latitude and longitude to null values
                        exifInfo.GPSData.Altitude = null;
                        exifInfo.GPSData.Latitude = null;
                        exifInfo.GPSData.LatitudeRef = null;
                        exifInfo.GPSData.Longitude = null;
                        exifInfo.GPSData.LongitudeRef = null;
                    }

                    // and update file
                    MetadataUtility.UpdateMetadata(path, exifMetadata);
                }
            }
            Console.WriteLine("Press any key to exit.");
        }
    }
    //ExEnd:PhotoCleaner
}
